Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/orientation.h

```diff
@@ -535,6 +449,819 @@ void orient(TriangleMesh& tm)
+ * - a non-closed surface component is reported as an isolated volume ignoring any inclusion test
+ * - a self-intersecting surface component is reported as an isolated volume ignoring any inclusion test
+ * - a surface component intersecting another surface component
+ *   is reported as an isolated volume, and so are all the surface components inside its
+ *   enclosed volume.
+ * - if `do_orientation_tests` is `true`, if the holes are not all equally oriented
+ *   (all inward or all outward) or if the holes and the outer boundary are equally
+ *   oriented, each surface component is reported as an isolated volume,
+ *   and so are all the surface components inside the corresponding enclosed volumes.
+ * - If all the faces of `tm` are part of the same surface component, then
+ *   no further test (open/closed, self-intersecting) is done and all faces
+ *   are assigned to the same volume components.
+ * - If `do_orientation_tests` is set to `true` and the surface components that are
+ *   outside all enclosed volumes are inward oriented, they are then considered as holes
+ *   of the unbounded volume (that has no outer boundary)
+ *
+ * A property map for each of `CGAL::face_index_t` and `CGAL::vertex_point_t`
+ * must be either available as an internal property map
+ * of `tm` or provided as one of the \ref pmp_namedparameters "Named Parameters".
+ *
+ * @tparam TriangleMesh a model of `FaceListGraph`
+ * @tparam FaceIndexMap a model of `WritablePropertyMap` with
+ *                      `boost::graph_traits<PolygonMesh>::%face_descriptor` as key type and
+ *                      `boost::graph_traits<PolygonMesh>::%faces_size_type` as value type.
+ * @tparam NamedParameters a sequence of \ref pmp_namedparameters "Named Parameters"
+ *
+ * @param tm the input triangle mesh
+ * @param volume_id_map the property map filled by this function with indices of volume components associated to the faces of `tm`
+ * @param np optional sequence of \ref pmp_namedparameters "Named Parameters" among the ones listed below
+ *
+ * \cgalNamedParamsBegin
+ *   \cgalParamBegin{vertex_point_map}
+ *     the property map with the points associated to the vertices of `tm`.
+ *   \cgalParamEnd
+ *   \cgalParamBegin{face_index_map}
+ *     a property map containing a unique id per face of `tm`, in the range `[0, num_faces(tm)[`.
+ *   \cgalParamEnd
+ *   \cgalParamBegin{face_connected_component_map}
+ *     a property map filled by this function and that will contain for each face the id
+ *     of its surface component in the range `[0, number of surface components - 1[`.
+ *   \cgalParamEnd
+ *   \cgalParamBegin{volume_inclusions}
+ *     a `reference_wrapper` (either from `boost` or the standard library) containing
+ *     a reference to an object that must be a model of the `BackInsertionSequence` concept,
+ *     with a value type being a model of `BackInsertionSequence` of `std::size_t`,
+ *     both types having the functions `reserve()` and `push_back()`.
```

Comment:
I'm adding a reserve() that will work for containers that do not have reserve, in https://github.com/CGAL/cgal/pull/4418/files#diff-f1e74bf64644af37d70f782ac4cb1ee9. Maybe use that instead?

Reply:
same remark as above